frontend/src/_helpers/appUtils.js

```diff
@@ -81,9 +81,33 @@ export function onComponentOptionChanged(_ref, component, option_name, value) {
   let componentData = components[componentName];
   componentData = componentData || {};
   componentData[option_name] = value;
-  useCurrentStateStore.getState().actions.setCurrentState({
-    components: { ...components, [componentName]: componentData },
+  let isListView = false,
+    isForm = false;
+  Object.keys(_ref.state.appDefinition.pages[_ref.state.currentPageId]?.components).forEach((item) => {
+    if (component.parent == item) {
+      try {
+        isListView =
+          _ref.state.appDefinition.pages[_ref.state.currentPageId]?.components[item]?.component?.component ===
+          'Listview';
+        isForm =
+          _ref.state.appDefinition.pages[_ref.state.currentPageId]?.components[item]?.component?.component === 'Form';
+      } catch {
```

Comment:
This will affect the performance if we have a lot of component and might be solved with different approach. 
We can check if the parent is `listView` or `Form` by using the parent ID in `Box.jsx` and the function call can be avoided